contracts/staking/GoodCompoundStaking.sol

```diff
@@ -37,47 +39,73 @@ contract GoodCompoundStaking is SimpleStaking {
 			_tokenSymbol,
 			_maxRewardThreshold
 		)
-	{}
-
+	{
+        tokenUsdOracle = _tokenUsdOracle;
+    }
+    // Address of the TOKEN/USD oracle from chainlink
+	address public tokenUsdOracle;
 	/**
-	 * @dev stake some DAI
-	 * @param _amount of dai to stake
+	 * @dev stake some Token
+	 * @param _amount of Token to stake
 	 */
 	function mintInterestToken(uint256 _amount) internal override {
 		cERC20 cToken = cERC20(address(iToken));
 		uint256 res = cToken.mint(_amount);
 
-		if (
-			res > 0
-		) //cDAI returns >0 if error happened while minting. make sure no errors, if error return DAI funds
-		{
-			require(res == 0, "Minting cDai failed, funds returned");
-		}
+	    require(res == 0, "Minting cToken failed, funds returned");
+	
 	}
 
 	/**
-	 * @dev redeem DAI from compound
-	 * @param _amount of dai to redeem
+	 * @dev redeem Token from compound
+	 * @param _amount of token to redeem in Token
 	 */
 	function redeem(uint256 _amount) internal override {
 		cERC20 cToken = cERC20(address(iToken));
-		require(cToken.redeemUnderlying(_amount) == 0, "Failed to redeem cDai");
+		require(cToken.redeemUnderlying(_amount) == 0, "Failed to redeem cToken");
 	}
 
+    /**
+     * @dev Function to redeem cToken for DAI
+     * @dev _amount of token in iToken
+     * @return return address of the DAI and amount of the DAI
+     */
 	function redeemUnderlying(uint256 _amount)
 		internal
 		override
 		returns (address, uint256)
 	{
-		uint256 tokenBalance = token.balanceOf(address(this));
-		cERC20 cToken = cERC20(address(iToken));
-		require(cToken.redeem(_amount) == 0, "Failed to redeem cDai");
-		uint256 redeemedAmount = token.balanceOf(address(this)) - tokenBalance;
-		return (address(token), redeemedAmount);
+
+        uint256 tokenBalance = token.balanceOf(address(this));
```

Comment:
we can optimize this a little.
no need for tokenBalance calculations as the correct state of this contract is to have 0 of token. this contract only holds iTokens

Reply:
Changed it as just send DAI balance of token as redeemed